### PR TITLE
Create time_bnds variable for CFTimeIndex or when resampling

### PR DESCRIPTION
**Pull request by jwenfai**
Find the time bounds (start and end times) for periods. In lieu of actual periods, CFTimeIndex indices, which are cftime.datetime objects, are used.

- [x] This PR addresses an already opened issue (for issue #74)
- [x] Tests for the changes have been added
- [x] Docs have been added / updated (docstrings have been added for the new functions, don't know if additional documentation is needed for the xclim project)

AFAIK, no breaking changes.

**Important Notes:**
1. Needs xarray version with CFTimeIndex resampling and QuarterOffset implemented. All tests are passing on my fork of xarray which has both features implemented.
2. Time bounds for indices with greater-than-day frequencies which start at the beginning of a month (frequencies with an 'S' suffix, e.g., 'MS', '7QS-JUN', 'AS-APR') are not supported. This is in keeping with pandas behavior. To see this for yourself, try doing `random_pandas_date_range.to_period('MS')`. Also you can try doing a conversion `pd.date_range('2000-01-01', periods=13, freq='MS').to_period()` -- notice that `freq` is converted to `M`.

I ran out of time so the code is not well integrated into `utils.py` and `test_utils.py`. Feel free to clean up the code so it conforms better to the xclim project's standards before merging.

Usage:
```python
index = xr.cftime_range(start='2000-01-01', periods=3,
                        freq='2QS', calendar='360_day')
time_bnds(index, '2Q')
Out[98]: 
((cftime.Datetime360Day(2000, 1, 1, 0, 0, 0, 0, 1, 1),
  cftime.Datetime360Day(2000, 3, 30, 23, 59, 59, 999999, 0, 91)),
 (cftime.Datetime360Day(2000, 7, 1, 0, 0, 0, 0, 6, 181),
  cftime.Datetime360Day(2000, 9, 30, 23, 59, 59, 999999, 5, 271)),
 (cftime.Datetime360Day(2001, 1, 1, 0, 0, 0, 0, 4, 1),
  cftime.Datetime360Day(2001, 3, 30, 23, 59, 59, 999999, 3, 91)))
da = xr.DataArray(range(100, 100+len(index)),
                  coords={'time': index}, dims='time')
time_bnds(da.resample(time='3M'), '3M')
Out[99]: 
((cftime.Datetime360Day(2000, 1, 1, 0, 0, 0, 0, 1, 1),
  cftime.Datetime360Day(2000, 3, 30, 23, 59, 59, 999999, 0, 91)),
 (cftime.Datetime360Day(2000, 4, 1, 0, 0, 0, 0, 0, 91),
  cftime.Datetime360Day(2000, 6, 30, 23, 59, 59, 999999, 6, 181)),
 (cftime.Datetime360Day(2000, 7, 1, 0, 0, 0, 0, 6, 181),
  cftime.Datetime360Day(2000, 9, 30, 23, 59, 59, 999999, 5, 271)),
 (cftime.Datetime360Day(2000, 10, 1, 0, 0, 0, 0, 5, 271),
  cftime.Datetime360Day(2000, 12, 30, 23, 59, 59, 999999, 4, 1)),
 (cftime.Datetime360Day(2001, 1, 1, 0, 0, 0, 0, 4, 1),
  cftime.Datetime360Day(2001, 3, 30, 23, 59, 59, 999999, 3, 91)))
```

### CODE REVIEW

**Comment by Zeitsperre** [comment]
@huard it looks like xarray has added some of the needed features in order for this PR to pass. Would it be alright if I tried a quick merge and update?

**Comment by Zeitsperre** [comment]
@huard 

Everything looks good from my end. The tests could be rewritten to be consistent with our current conventions, but they are all passing. Feel free to review.

File: tests/test_utils.py [hunk 1]
```diff
@@ -660,6 +660,52 @@ def test_simple(self, tas_series):
...
+    cftime_ends = CFTimeIndex(cftime_ends).to_datetimeindex()
+    # cftime resolution goes down to microsecond only, code below corrects
+    # that to allow for comparison with pandas datetime
+    cftime_ends += np.timedelta64(999, "ns")
+    datetime_starts = da_datetime._full_index.to_period(freq).start_time
+    datetime_ends = da_datetime._full_index.to_period(freq).end_time
+    print(cftime_starts)
+    print(datetime_starts)
+    print(cftime_ends)
+    print(datetime_ends)
```
Comment (huard) [inline tests/test_utils.py]: Remove print statements

File: HISTORY.rst [hunk 2]
```diff
@@ -4,12 +4,13 @@ History
 
 0.14.x
 ------
+* Creation of `time_bnds` variables when resampling data encoded with `CFTimeIndex` (non-standard calendars).
```
Comment (huard) [inline HISTORY.rst]: At the moment, the functionalities are not used by the Indicator class. So I would change that to *Added utility functions to compute `time_bnds`*.

Comment (Zeitsperre) [reply HISTORY.rst]: Want me to reopen this PR and add the change? Alternatively, we could just hotfix that line on master. What do you think?

Comment (huard) [reply HISTORY.rst]: No problem with hotfix.

**Pull request merged**